tools/check-openapi

```diff
@@ -43,15 +43,15 @@ async function checkFile(file) {
     const promises = [];
 
     visit(doc, {
-        Map(key, node) {
+        Map(_key, node) {
```

Comment:
Can we use `ignored_key` as the naming convention? I've been pretty happy with using that in the server for similar things, and I think it feels more readable and harder to misinterpret than just the `_` prefix.

Reply:
Python is a different language where the `_` prefix has a [special meaning](https://docs.python.org/3.1/reference/simple_stmts.html#the-import-statement) to the `import` statement and therefore [conventionally](https://peps.python.org/pep-0008/#descriptive-naming-styles) indicates “internal use” rather than “unused”. JavaScript has real private members and a module system that can actually encapsulate things, so it has different conventions. Trying to pretend our JavaScript is Python won’t make our code more readable.

The leading underscore convention is [built into TypeScript](https://github.com/microsoft/TypeScript/pull/9464).

In Python, we use an `ignored_` prefix for lots of things, most of which aren’t unused variables.

<details>

```
used    tools/lib/capitalization.py:    ignored_phrase = matchobj.group(0)
used    tools/lib/capitalization.py:    safe_string = get_safe_phrase(ignored_phrase)
UNUSED  zerver/actions/message_edit.py:    message, ignored_user_message = access_message(user_profile, message_id)
UNUSED  zerver/lib/email_notifications.py:    for bucket_tup, ignored_max_id in bucket_tups:
used    zerver/lib/request.py:            ignored_parameters = set(
used    zerver/lib/request.py:            if ignored_parameters:
used    zerver/lib/request.py:                    ignored_parameters
used    zerver/lib/test_classes.py:        ignored_parameters: Optional[List[str]] = None,
used    zerver/lib/test_classes.py:        if ignored_parameters is None:
used    zerver/lib/test_classes.py:            for param in ignored_parameters:
used    zerver/tests/test_bots.py:        response_dict = self.assert_json_success(result, ignored_parameters=["method"])
used    zerver/tests/test_decorators.py:        self.assert_json_success(result, ignored_parameters=["age"])
used    zerver/tests/test_decorators.py:        ignored_parameters = ["age", "dies", "location"]
used    zerver/tests/test_decorators.py:        json_result = self.assert_json_success(result, ignored_parameters=ignored_parameters)
used    zerver/tests/test_decorators.py:        self.assert_json_success(result, ignored_parameters=["author"])
used    zerver/tests/test_realm.py:        self.assert_json_success(result, ignored_parameters=["emoji_set"])
used    zerver/tests/test_settings.py:        self.assert_json_success(result, ignored_parameters=["invalid_setting"])
used    zerver/tests/test_subs.py:        self.assert_json_success(result, ignored_parameters=["invalid_parameter"])
used    zerver/tests/test_users.py:        self.assert_json_success(result, ignored_parameters=["short_name"])
UNUSED  zerver/views/home.py:            (narrow_stream, ignored_sub) = access_stream_by_name(user_profile, narrow_stream_name)
UNUSED  zerver/views/message_edit.py:    message, ignored_user_message = access_message(user_profile, message_id)
UNUSED  zerver/views/message_edit.py:    message, ignored_user_message = access_message(user_profile, message_id, lock_message=True)
UNUSED  zerver/views/streams.py:    stream, ignored_sub = access_stream_by_id(user_profile, stream_id)
UNUSED  zerver/views/users.py:        (default_sending_stream, ignored_sub) = access_stream_by_name(
UNUSED  zerver/views/users.py:        (default_events_register_stream, ignored_sub) = access_stream_by_name(
used    zerver/webhooks/github/tests.py:        ignored_actions = [
used    zerver/webhooks/github/tests.py:        for action in ignored_actions:
used    zerver/webhooks/github/tests.py:        ignored_actions = [
used    zerver/webhooks/github/tests.py:        for action in ignored_actions:
used    zerver/webhooks/github/tests.py:        ignored_events = [
used    zerver/webhooks/github/tests.py:        for event in ignored_events:
used    zerver/webhooks/jira/tests.py:        ignored_actions = [
used    zerver/webhooks/jira/tests.py:        for action in ignored_actions:
used    zerver/webhooks/trello/view/card_actions.py:        ignored_fields = [
used    zerver/webhooks/trello/view/card_actions.py:        for field in ignored_fields:
```

</details>

Reply:
Ahh, I guess I was thinking about return values of functions, not function parameters, for the thing we have a stronger convention on. With function parameters, the `ignored_` convention feels more natural to me.